pkg/syncer/spec/dns/networkpolicy_dns.yaml

```diff
@@ -0,0 +1,45 @@
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: Name
+  namespace: Namespace
+spec:
+  podSelector:
+    matchLabels:
+      app: Name
+  policyTypes:
+    - Ingress
+    - Egress
+  ingress:
+    - from:
+        - namespaceSelector:
+            matchLabels:
+              internal.workload.kcp.io/cluster: Cluster
```

Comment:
So here you're finally matching the namespaces based on the `SyncTargetKey` here, which is not enough.
We should only allow traffic from namespaces that originate from the KCP workspace associated to this DNS Pod.

It seems to me that we need a new label here: a **_tenant ID_**, that uniquely identifies a KCP tenant (==syncTarget + KCP workspace) in the workload cluster.
Something like what I had setup in the (non-yet-merged) DNS-cleanup PR: https://github.com/kcp-dev/kcp/pull/2455/files#diff-c6d132e1ca7f7d7849896191d00c73d4cf74bfab6cad41ad327de656c3585d95R86

Reply:
The reason I used Cluster is because in some places cluster = workspace (at least here https://github.com/kcp-dev/kcp/blob/release-0.10/pkg/syncer/spec/spec_process.go#L113). Probably what's is missing is a developer guide describing the dlevel terminology. Documenting which labels are currently used would also helps determining if `tenantID` is right.